### ISSUE TITLE
Writing new standards documentation

### ISSUE THREAD

**Issue by tassoman** [squizlabs/HTML_CodeSniffer#252]
Hello there, I'd like to start writing a new standard based on [Criteri e metodi per la verifica tecnica e requisiti tecnici di accessibilità previsti dalla legge 9 gennaio 2004, n. 4.](https://www.agid.gov.it/it/DM-8-luglio-2005-Allegato-A) :smile: 

Looks like something difficult but it's simply based on the four principles of WCAG2.0 but structured in 12 requirements and sub-requirements.

For example sub-requirement 2.1 is based on WCAG2.0 » 1.2.1: 

> Punto di controllo 2.1 - Contenuti registrati presentati in formato solo audio, solo video o animazione senza audio
> Riferimento WCAG 2.0: Criterio di successo 1.2.1

Can you please provide an example or explain how to accomplish this writing?

**Comment by ironikart** [comment]
Hi @tassoman 

Firstly I should warn that this tool was not written with the idea that custom rules would be added.

However, if you want to customise it this way you would achieve it by:
* Add directory to ./Standards`
* create a `ruleset.js` file (can base it from the existing ones). This needs a unique name in the _global namespace.
* Run `grunt build` to generate a new version of `build/HTMLCS.js` (All the js files in `./Standards` will be included in the build).

**Comment by tassoman** [comment]
Thank you for caring about it, I'm done adding the Standard's structure.
I ask you some informations about `getMsgInfo()` method.
Targeting on HTMLCS_WCAG2AAA object, I've added my twelve principles. Now I'd like to build my crit list, and linking existing sniffs.

Following the example I mentioned early in the issue, I need to create my `successCritList 2.1` and link to sniffs for `successCritList.1.2.1` in WCAG2AAA.

So that i suppose to write:

```javascript
{
  var successCritList = {
    '2.1': {
      name: 'Audio-only and Video-only (Prerecorded)',
      landmark: 'media-equiv-av-only-alt',
      priority: 1,
    },
  } // end of crit-list
} // end of global obj
```
In the end, how do I link that sniffs to my criteria? Thank you for your time

**Comment by ironikart** [comment]
That success criteria list exists only in getMsgInfo() and is used to [generate a HTML link to it's documentation](https://github.com/squizlabs/HTML_CodeSniffer/blob/5ca0b9b0accb597faaeb13f27e7c961d2a3d301f/Standards/WCAG2AAA/ruleset.js#L439). If you want to generate your own links to documentation (e.g. agid.gov.it) then you just need to figure out a way to turn the `code` variable passed to getMsgInfo() into that link, you don't necessarily need to follow the same content as the existing function, just use it as a rough guide (as long as you take a code and return an array of html strings).
